### ISSUE TITLE
Property Agentic AI

### ISSUE THREAD

**Issue by sahirmaharaj** [microsoft/fabcon-global-hack-25#20]
### First and Last Name

Sahir Maharaj

### MS Learn Fabric Skilling Plan Completion

Yes

### How It All Started

It began with a pattern I couldn’t ignore.

Most property calls don’t happen during office hours - they happen late at night, on weekends, or while people are on the move.

But that’s exactly when most real-estate offices are closed.
Prospective buyers hit voicemails, long queues, or generic chat menus that feel cold and robotic.

And, by the time an agent calls back, that spark of interest is often gone.

After talking to many people in the real estate space (especially in the UAE) one thing became clear to me - slow responses and poor engagement are costing businesses real money.

Leads go cold, deals fall through, and data from thousands of calls just sits unused. 

That’s what inspired me to build a solution to solve it - an AI-powered voice concierge that can actually talk, understand, and respond naturally, no matter the time of day.

### What I’ve Built

This AI agent is like having your best real estate agent available 24/7 - friendly, multilingual, and always ready to help.

It instantly answers calls, greets prospects by name, and provides accurate, real-time information about listings, floor plans, or neighborhood details.

And it doesn’t stop there - I built it to ask qualifying questions, book viewings in CRM tools like Zoho, and connect to a human agent when needed.

> This way, every call becomes valuable data.

The system logs conversations in an Azure SQL warehouse, runs analysis through Microsoft Fabric Data Science notebooks, and uses NLP models to detect sentiment, intent, and even predict buyer behavior.

The more it interacts, the smarter it gets.

Right now, it’s still in active development, as I’m refining the conversational flow, improving model accuracy, and adding integrations based on pilot feedback.

Over the next few months, the focus is to make **every** conversation more human and every insight more actionable.

### The Tech Behind It

When it came to choosing the tech stack, I deliberately chose Microsoft Fabric as the foundation.

I genuinely believe it’s one of the most powerful ecosystems for building AI solutions end-to-end... and more people should know what it’s capable of.

Most of the heavy lifting happens here, where I’ve built the data pipelines, ML models, and Power BI reports.

The core runs on Fabric Data Science notebooks and Dataflow Gen2, storing everything in a Lakehouse environment that connects smoothly to other Azure components.

The stack is a mix of machine learning and AI technologies:
- Custom ML Models for classification and forecasting (CatBoost, SVM, CNNs, XGBoost, LightGBM).
- LangChain and embedding techniques for smart contextual retrieval.
- GPT-4o-mini-high and Gemini models for adaptive dialogue generation.
- Vonage API for realistic, cloned voice interactions.
- Fabric Data Science notebooks for advanced NLP and tone analytics.
- Microsoft Power BI for reporting that surface lead quality and agent performance.
- Twilio for mobile calling, routing, and human-agent handoff when needed.

> About 75% of the build sits within Azure and Microsoft Fabric.

Everything else including custom webhooks and API orchestration runs on my own server.

### Why It Matters

The real estate industry is one of the biggest spenders on customer engagement (over $300 billion a year globally), yet it still loses up to 40% of leads due to human error or slow response.

And in multilingual markets like Dubai, where buyers and tenants speak across multiple dialects, that challenge multiplies.

My solution aims to bridge that gap.

> I’ve already been in discussions with executive stakeholders in the UAE that’s exploring a pilot implementation.

The early response has been encouraging as they see the potential to cut operational costs, handle more inbound leads, and deliver a better customer experience overall.

### Roadmap Ahead

And this is just the beginning.

Soon I plan to have a pilot rollout with mid-sized UAE brokerages.

Here, I’ll fine-tune the voice recognition models, refine the CRM connections, and gather data to make the conversations even smoother.

> The goal is to eventually expand to larger companies and integrate with other CRM systems, where I’ll also introduce deeper analytics layers.

Even as the product evolves, the **vision stays the same** - to transform how real estate companies engage with people.

### How I will use the prize money

Upon winning any of the monetary prizes, it’s going straight back into the project.

The next big milestone is training more advanced models on Azure - especially around emotional understanding, voice tone, and multilingual sentiment prediction.

The goal is to make each AI conversation sound more natural, more contextual, and ultimately more helpful.

> This is more than just a hackathon experiment, but a real solution to a real problem.

Because buying a home isn’t _just transactional_, it’s emotional.

And I believe that the MORE human we make these digital experiences, the stronger the connection :)

<img width="2560" height="1600" alt="Image" src="https://github.com/user-attachments/assets/22fe01cf-726a-4030-8059-f9d4dc1f6b88" />

<img width="2560" height="1600" alt="Image" src="https://github.com/user-attachments/assets/932eb228-0198-4ce6-8ee4-04979a69a9f9" />

<img width="2560" height="1600" alt="Image" src="https://github.com/user-attachments/assets/32eed2b2-8773-4927-b861-e69861cbb610" />

<img width="2560" height="1600" alt="Image" src="https://github.com/user-attachments/assets/50d19f2a-c531-4cc3-a463-d1f2836bbfba" />

<img width="2560" height="1600" alt="Image" src="https://github.com/user-attachments/assets/95f748c6-c1e9-45d7-9630-51790c8e664a" />

<img width="2560" height="1600" alt="Image" src="https://github.com/user-attachments/assets/7f5f963d-0feb-4609-8056-c1f12b1d5fee" />

<img width="2560" height="1600" alt="Image" src="https://github.com/user-attachments/assets/fb9ac05a-56c4-467b-8785-8f1e4ab7d7e8" />

### Project Repository URL

https://github.com/sahirmaharaj/ai-agent

### Project Demo video URL

https://youtu.be/Po7Fhs01Jy4

### Registration status.

- [x] I have completed registration
